commit 64e9bbdd9588ad8b74c687a192540ba847895706
Author: Ramalingam C <ramalingam.c@intel.com>
Date:   Thu Feb 21 23:41:56 2019 +0530

    misc/mei/hdcp: Client driver for HDCP application
    
    ME FW contributes a vital role in HDCP2.2 authentication.
    HDCP2.2 driver needs to communicate to ME FW for each step of the
    HDCP2.2 authentication.
    
    ME FW prepare and HDCP2.2 authentication  parameters and encrypt them
    as per spec. With such parameter Driver prepares HDCP2.2 auth messages
    and communicate with HDCP2.2 sink.
    
    Similarly HDCP2.2 sink's response is shared with ME FW for decrypt and
    verification.
    
    Once All the steps of HDCP2.2 authentications are complete on driver's
    request ME FW will configure the port as authenticated and supply the
    HDCP keys to the Gen HW for encryption.
    
    Only after this stage HDCP2.2 driver can start the HDCP2.2 encryption
    for a port.
    
    ME FW is interfaced to kernel through MEI Bus Driver. To obtain the
    HDCP2.2 services from the ME FW through MEI Bus driver MEI Client
    Driver is developed.
    
    v2:
      hdcp files are moved to drivers/misc/mei/hdcp/ [Tomas]
    v3:
      Squashed the Kbuild support [Tomas]
      UUID renamed and Module License is modified [Tomas]
      drv_data is set to null at remove [Tomas]
    v4:
      Module name is changed to "MEI HDCP"
      I915 Selects the MEI_HDCP
    v5:
      Remove redundant text from the License header
      Fix malformed licence
      Removed the drv_data resetting.
    v6:
      K-Doc addition. [Tomas]
    v7:
      %s/UUID_LE/GUID_INIT [Tomas]
      GPL Ver is 2.0 than 2.0+ [Tomas]
    v8:
      Added more info into Kconfig addition [Tomas]
    
    Signed-off-by: Ramalingam C <ramalingam.c@intel.com>
    Signed-off-by: Tomas Winkler <tomas.winkler@intel.com>
    Acked-by: Tomas Winkler <tomas.winkler@intel.com>
    Signed-off-by: Daniel Vetter <daniel.vetter@ffwll.ch>
    Link: https://patchwork.freedesktop.org/patch/msgid/1550772730-23280-3-git-send-email-ramalingam.c@intel.com

diff --git a/drivers/misc/mei/Kconfig b/drivers/misc/mei/Kconfig
index c49e1d2269af..74e2c667dce0 100644
--- a/drivers/misc/mei/Kconfig
+++ b/drivers/misc/mei/Kconfig
@@ -43,3 +43,13 @@ config INTEL_MEI_TXE
 
 	  Supported SoCs:
 	  Intel Bay Trail
+
+config INTEL_MEI_HDCP
+	tristate "Intel HDCP2.2 services of ME Interface"
+	select INTEL_MEI_ME
+	depends on DRM_I915
+	help
+	  MEI Support for HDCP2.2 Services on Intel platforms.
+
+	  Enables the ME FW services required for HDCP2.2 support through
+	  I915 display driver of Intel.